src/polyswarm_api/__main__.py

```diff
@@ -214,31 +215,52 @@ def url_scan(ctx, url, url_file, force, timeout):
     ctx.obj['output'].write(str(rf))
 
 
-@click.option('-r', '--hash-file', help='File of hashes, one per line.', type=click.File('r'))
-@click.option('--hash-type', help='Hash type to search [sha256|sha1|md5], default=sha256', default='sha256')
-@click.argument('hash', nargs=-1, callback=validate_hash)
-@polyswarm.command('search', short_help='search for hash')
+@click.option('-r', '--hash-file', help="File of hashes, one per line.", type=click.File('r'))
+@click.option("--hash-type", help="Hash type to search [sha256|sha1|md5], default=sha256", default="sha256")
+@click.argument('search_arguments', nargs=-1)
+@polyswarm.command("search", short_help="search for hashes separated by space or JSON query")
 @click.pass_context
-def search(ctx, hash, hash_file, hash_type):
+def search(ctx, search_arguments, hash_file, hash_type):
     """
     Search PolySwarm for files matching sha256 hashes
     """
-    api = ctx.obj['api']
+    def _is_json(string):
```

Comment:
I am usually a fan of "[easier to ask forgiveness than permission](https://docs.python.org/3/glossary.html#term-eafp)" way of programming. I would try to just load and use assuming a default flow as `json` and if it fails, process the alternative flow. Just a style comment though.

Reply:
I tried to re-use the same "search" argument for both hash and query searches, so if it's json, I assume is a query search, otherwise (not json) a hash search. An option would be handling this in a try / catch block, but I think this is cleaner (not by much though, I might be even wrong).